src/test/kotlin/com/msd/robot/application/RobotApplicationServiceTest.kt

```diff
@@ -169,4 +170,40 @@ class RobotApplicationServiceTest {
         assertEquals(planet2, robot1.planet)
         verify(exactly = 1) { robotRepository.save(robot1) }
     }
+
+    @Test
+    fun `Unknown robotId when regenerating causes an exception to be thrown`() {
+        // given
+        every { robotRepository.findByIdOrNull(unknownRobotId) } returns null
+        // then
+        assertThrows<RobotNotFoundException> {
+            robotApplicationService.regenerateEnergy(RegenCommand(unknownRobotId, UUID.randomUUID()))
+        }
+    }
+
+    @Test
+    fun `playerId not matching ownerId when regenerating causes an exception to be thrown`() {
+        // given
+        robot1.move(Planet(UUID.randomUUID()), 10)
+        every { robotRepository.findByIdOrNull(robot1.id) } returns robot1
+
+        // then
+        assertThrows<InvalidPlayerException> {
+            robotApplicationService.regenerateEnergy(RegenCommand(robot1.id, UUID.randomUUID()))
+        }
```

Comment:
Also assert that the energy hasn't increased. This makes sure that future changes don't disrupt the order of checks